standalone/pruntime/enclave/src/system/gk.rs

```diff
@@ -132,7 +127,44 @@ where
         self.egress.set_dummy(true);
     }
 
+    pub fn master_pubkey_uploaded(&mut self) {
+        self.master_pubkey_on_chain = true;
+    }
+
+    pub fn share_master_key(&mut self, pubkey: &WorkerPublicKey, ecdh_pubkey: &EcdhPublicKey) {
+        info!("Gatekeeper: try dispatch master key");
+        let derived_key = self
+            .master_key
+            .derive_sr25519_pair(&[&crate::generate_random_info()])
+            .expect("should not fail with valid info; qed.");
+        let my_ecdh_key = derived_key
+            .derive_ecdh_key()
+            .expect("ecdh key derivation should never failed with valid master key; qed.");
+        let secret = ecdh::agree(&my_ecdh_key, &ecdh_pubkey.0)
+            .expect("should never fail with valid ecdh key; qed.");
+        let iv = crate::generate_random_iv();
```

Comment:
To make it deterministic, we should derive the IV from a combination of master key and the block height

Reply:
fixed